range.c

```diff
@@ -1880,13 +1983,40 @@ range_alloc(VALUE klass)
 
 /*
  *  call-seq:
- *     range.count                 -> int
- *     range.count(item)           -> int
- *     range.count { |obj| block } -> int
+ *    count -> integer
+ *    count(&method) -> integer
+ *    count(object) -> integer
+ *    count {|element| ... } -> integer
+ *
+ *  Returns the count of elements, based on an argument or block criterion, if given.
+ *
+ *  With no argument and no block given, returns the number of elements:
+ *
+ *    (1..4).count      # => 4
+ *    (1...4).count     # => 3
+ *    ('a'..'d').count  # => 4
+ *    ('a'...'d').count # => 3
+ *    (1..).count       # => Infinity
+ *    (..4).count       # => Infinity
+ *
+ *  With argument +method+, returns the number of elements for which
+ *  <tt>method(element)</tt> returns a truthy value:
+ *
+ *    (1..4).count(&:even?) # => 2
+ *
+ *  With argument +object+, returns the number of +object+ found in +self+,
+ *  which will always be zero or one:
```

Comment:
Change `always` to `usually`.  It's possible to construct a range where `count` with an argument can return other results:

```ruby
S = Struct.new(:i) do
  include Comparable
  def <=>(other)
    i <=> other.i
  end
  def succ
    if @succ
      self.class.new(i+1)
    else
      @succ = true
      self
    end
  end
end

(S.new(1)..S.new(2)).count(S.new(1))
```

Reply:
Done.